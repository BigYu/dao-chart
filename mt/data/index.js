module.exports = async function(ctx, next) {
  const { name, parameters } = ctx.request.body;
  const handle = require(`./${name}-handler.js`);

  ctx.body = await handle(parameters);
  await next();
};
