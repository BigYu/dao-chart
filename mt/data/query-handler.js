module.exports = function(parameters) {
  return { parameters };
};
