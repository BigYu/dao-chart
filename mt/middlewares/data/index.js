const Promise = require('bluebird');
const Boom = require('boom');
const _ = require('lodash');

async function handleSingleQuery(query) {
  const { name, parameters } = query;

  if (_.isEmpty(name) || !_.isString(name)) {
    throw Boom.badRequest('Missing "name"');
  }

  if (_.isEmpty(parameters) || !_.isObject(parameters)) {
    throw Boom.badRequest('Missing "parameters"');
  }

  const handle = require(`./handlers/${name}.js`);

  return handle(parameters);
}

module.exports = () => async function(ctx, next) {
  const { body } = ctx.request;

  if (_.isArray(body)) {
    ctx.body = await Promise.map(body, async q => handleSingleQuery(q));
    await next();
  } else {
    ctx.body = await handleSingleQuery(ctx.request.body);
    await next();
  }
};
