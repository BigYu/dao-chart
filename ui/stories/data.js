export const source = [
  ['product', '2015', '2016', '2017'],
  ['Matcha Latte', 43.3, 85.8, 93.7],
  ['Milk Tea', 83.1, 73.4, 55.1],
  ['Cheese Cocoa', 86.4, 65.2, 82.5],
  ['Walnut Brownie', 72.4, 53.9, 39.1],
];

export const pieSource = [
  ['name', 'value'],
  ['直接访问', 335],
  ['邮件营销', 310],
  ['联盟广告', 234],
  ['视频广告', 135],
  ['搜索引擎', 1548],
];
