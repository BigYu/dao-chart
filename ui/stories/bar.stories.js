import React from 'react';
import { storiesOf } from '@storybook/react';
import Bar from '../src/component/bar';
import { source } from './data';

storiesOf('Bar Chart', module)
  .add('Simple bar', () => (
    <Bar source={source} />
  ));
