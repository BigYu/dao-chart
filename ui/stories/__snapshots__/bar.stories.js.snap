// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Simple bar 1`] = `
<EchartsReact
  className=""
  echarts={Object {}}
  lazyUpdate={false}
  loadingOption={null}
  notMerge={false}
  onChartReady={[Function]}
  onEvents={Object {}}
  option={
    Object {
      "dataset": Object {
        "dimensions": Array [
          "product",
          "2015",
          "2016",
          "2017",
        ],
        "source": Array [
          Array [
            "product",
            "2015",
            "2016",
            "2017",
          ],
          Array [
            "Matcha Latte",
            43.3,
            85.8,
            93.7,
          ],
          Array [
            "Milk Tea",
            83.1,
            73.4,
            55.1,
          ],
          Array [
            "Cheese Cocoa",
            86.4,
            65.2,
            82.5,
          ],
          Array [
            "Walnut Brownie",
            72.4,
            53.9,
            39.1,
          ],
        ],
      },
      "legend": Object {},
      "series": Array [
        Object {
          "name": "2015",
          "type": "bar",
        },
        Object {
          "name": "2016",
          "type": "bar",
        },
        Object {
          "name": "2017",
          "type": "bar",
        },
      ],
      "tooltip": Object {},
      "xAxis": Object {
        "type": "category",
      },
      "yAxis": Object {},
    }
  }
  opts={Object {}}
  showLoading={false}
  source={
    Array [
      Array [
        "product",
        "2015",
        "2016",
        "2017",
      ],
      Array [
        "Matcha Latte",
        43.3,
        85.8,
        93.7,
      ],
      Array [
        "Milk Tea",
        83.1,
        73.4,
        55.1,
      ],
      Array [
        "Cheese Cocoa",
        86.4,
        65.2,
        82.5,
      ],
      Array [
        "Walnut Brownie",
        72.4,
        53.9,
        39.1,
      ],
    ]
  }
  style={Object {}}
  theme={null}
/>
`;

exports[`Snapshot of Simple horizontal bar 1`] = `
<EchartsReact
  className=""
  echarts={Object {}}
  lazyUpdate={false}
  loadingOption={null}
  notMerge={false}
  onChartReady={[Function]}
  onEvents={Object {}}
  option={
    Object {
      "dataset": Object {
        "dimensions": Array [
          "product",
          "2015",
          "2016",
          "2017",
        ],
        "source": Array [
          Array [
            "product",
            "2015",
            "2016",
            "2017",
          ],
          Array [
            "Matcha Latte",
            43.3,
            85.8,
            93.7,
          ],
          Array [
            "Milk Tea",
            83.1,
            73.4,
            55.1,
          ],
          Array [
            "Cheese Cocoa",
            86.4,
            65.2,
            82.5,
          ],
          Array [
            "Walnut Brownie",
            72.4,
            53.9,
            39.1,
          ],
        ],
      },
      "legend": Object {},
      "series": Array [
        Object {
          "name": "2015",
          "type": "bar",
        },
        Object {
          "name": "2016",
          "type": "bar",
        },
        Object {
          "name": "2017",
          "type": "bar",
        },
      ],
      "tooltip": Object {
        "axisPointer": Object {
          "type": "shadow",
        },
        "trigger": "axis",
      },
      "xAxis": Object {
        "type": "value",
      },
      "yAxis": Object {
        "type": "category",
      },
    }
  }
  opts={Object {}}
  showLoading={false}
  source={
    Array [
      Array [
        "product",
        "2015",
        "2016",
        "2017",
      ],
      Array [
        "Matcha Latte",
        43.3,
        85.8,
        93.7,
      ],
      Array [
        "Milk Tea",
        83.1,
        73.4,
        55.1,
      ],
      Array [
        "Cheese Cocoa",
        86.4,
        65.2,
        82.5,
      ],
      Array [
        "Walnut Brownie",
        72.4,
        53.9,
        39.1,
      ],
    ]
  }
  style={Object {}}
  theme={null}
/>
`;

exports[`Snapshot of Simple ordered bar 1`] = `
<EchartsReact
  className=""
  echarts={Object {}}
  lazyUpdate={false}
  loadingOption={null}
  notMerge={false}
  onChartReady={[Function]}
  onEvents={Object {}}
  option={
    Object {
      "dataset": Object {
        "dimensions": Array [
          "framework",
          "star",
        ],
        "source": Array [
          Object {
            "framework": "Backbone",
            "star": 50,
          },
          Object {
            "framework": "React",
            "star": 1000,
          },
          Object {
            "framework": "Angularjs",
            "star": 1020,
          },
        ],
      },
      "legend": Object {},
      "series": Object {
        "name": "star",
        "type": "bar",
      },
      "tooltip": Object {
        "axisPointer": Object {
          "type": "shadow",
        },
        "trigger": "axis",
      },
      "xAxis": Object {
        "type": "value",
      },
      "yAxis": Object {
        "type": "category",
      },
    }
  }
  opts={Object {}}
  showLoading={false}
  source={
    Array [
      Array [
        "framework",
        "star",
      ],
      Array [
        "React",
        1000,
      ],
      Array [
        "Angularjs",
        1020,
      ],
      Array [
        "Backbone",
        50,
      ],
    ]
  }
  style={Object {}}
  theme={null}
/>
`;
